test/testfolderman.cpp

```diff
@@ -130,32 +132,32 @@ private slots:
 #endif
 
         // Invalid paths
-        QVERIFY(!folderman->checkPathValidityForNewFolder(QString()).isNull());
+        QVERIFY(!folderman->checkPathValidityForNewFolder({}, type, uuid).isNull());
 
 
         // REMOVE ownCloud2 from the filesystem, but keep a folder sync'ed to it.
         QDir(dirPath + QStringLiteral("/ownCloud2/")).removeRecursively();
-        QVERIFY(!folderman->checkPathValidityForNewFolder(dirPath + QStringLiteral("/ownCloud2/blublu")).isNull());
-        QVERIFY(!folderman->checkPathValidityForNewFolder(dirPath + QStringLiteral("/ownCloud2/sub/subsub/sub")).isNull());
+        QVERIFY(!folderman->checkPathValidityForNewFolder(dirPath + QStringLiteral("/ownCloud2/blublu"), type, uuid).isNull());
+        QVERIFY(!folderman->checkPathValidityForNewFolder(dirPath + QStringLiteral("/ownCloud2/sub/subsub/sub"), type, uuid).isNull());
 
         { // check for rejection of a directory with `.sync_*.db`
             QVERIFY(dir2.mkpath(QStringLiteral("db-check1")));
-            QVERIFY(folderman->checkPathValidityForNewFolder(dirPath + QStringLiteral("/db-check1")).isNull());
+            QVERIFY(folderman->checkPathValidityForNewFolder(dirPath + QStringLiteral("/db-check1"), type, uuid).isNull());
             QFile f(dirPath + QStringLiteral("/db-check1/.sync_something.db"));
             QVERIFY(f.open(QFile::Truncate | QFile::WriteOnly));
             f.close();
             QVERIFY(QFileInfo::exists(dirPath + QStringLiteral("/db-check1/.sync_something.db")));
-            QVERIFY(!folderman->checkPathValidityForNewFolder(dirPath + QStringLiteral("/db-check1")).isNull());
+            QVERIFY(!folderman->checkPathValidityForNewFolder(dirPath + QStringLiteral("/db-check1"), type, uuid).isNull());
         }
 
         { // check for rejection of a directory with `._sync_*.db`
             QVERIFY(dir2.mkpath(QStringLiteral("db-check2")));
-            QVERIFY(folderman->checkPathValidityForNewFolder(dirPath + QStringLiteral("/db-check2")).isNull());
+            QVERIFY(folderman->checkPathValidityForNewFolder(dirPath + QStringLiteral("/db-check2"), type, uuid).isNull());
             QFile f(dirPath + QStringLiteral("/db-check2/._sync_something.db"));
             QVERIFY(f.open(QFile::Truncate | QFile::WriteOnly));
             f.close();
             QVERIFY(QFileInfo::exists(dirPath + QStringLiteral("/db-check2/._sync_something.db")));
-            QVERIFY(!folderman->checkPathValidityForNewFolder(dirPath + QStringLiteral("/db-check2")).isNull());
+            QVERIFY(!folderman->checkPathValidityForNewFolder(dirPath + QStringLiteral("/db-check2"), type, uuid).isNull());
```

Comment:
Please add a test case for the scenario we are trying to solve here.

Reply:
done